common/model-views.cpp

```diff
@@ -1911,30 +1926,51 @@ namespace rs2
         return results;
     }
 
-    bool is_resolution_max(const std::vector<std::pair<int, int>>& res_values, int width, int height, const std::pair<int, int>& calibration_res)
+    bool subdevice_model::is_d405_oem_cal_profile(int d405_calibration_width, int d405_calibration_height) const
+    {
+        // checking format
+        bool is_cal_format = false;
+        for (auto it = stream_enabled.begin(); it != stream_enabled.end(); ++it)
+        {
+            if (it->second)
+            {
+                auto format = format_values.at(it->first)[ui.selected_format_id.at(it->first)];
+                if (format == RS2_FORMAT_Y16)
+                {
+                    is_cal_format = true;
+                    break;
+                }
+            }
+        }
+        if (is_cal_format)
+        {
+            // checking resolution
+            auto width = res_values[ui.selected_res_id].first;
+            auto height = res_values[ui.selected_res_id].second;
+            bool is_cal_res = (width == d405_calibration_width && height == d405_calibration_height);
+            return is_cal_format && is_cal_res;
+        }
+        return false;        
+    }
+
+    std::pair<int, int> subdevice_model::get_max_resolution(rs2_stream stream) const
     {
-        auto size = res_values.size();
-        int max_width = (res_values[size - 1].first != calibration_res.first) ? res_values[size - 1].first : res_values[size - 2].first;
-        int max_height = (res_values[size - 1].second != calibration_res.second) ? res_values[size - 1].second : res_values[size - 2].second;
-        return width == max_width && height == max_height;
+        if (resolutions_per_stream.count(stream) > 0)
+            return resolutions_per_stream.at(stream).back();
+        throw std::runtime_error("No such stream in this sensor");
     }
 
-    std::vector<stream_profile> subdevice_model::get_selected_profiles(bool throw_exception_flag)
+    std::vector<stream_profile> subdevice_model::get_selected_profiles(bool enforce_interstream_policies)
```

Comment:
rename to `enforce_inter_stream_policies`

Reply:
Done